crypto/cms/cms_ess.c

```diff
@@ -15,6 +15,9 @@
 #include <openssl/err.h>
 #include <openssl/cms.h>
 #include "cms_lcl.h"
+#include <openssl/ts.h>
+#include "../ts/ts_lcl.h"
```

Comment:
What functions in ts_lcl are needed?  You should put those declarations in a new header file that is in crypto/internal directory.

Reply:
`ESS_SIGNING_CERT_V2 `and `ESS_CERT_ID_V2`.
Yes, better move these structures to `cms_lcl.h` along with `CMS_SignerInfo `